varats/paper/artefacts.py

```diff
@@ -0,0 +1,297 @@
+    """
+    An artefact defining a :class:`plot<varats.plots.plot.Plot>`.
+
+    Args:
+        name: The name of this artefact.
+        output_path: The path where the plot this artefact produces will be
+                     stored.
+        plot_type: The
+                    :attr:`type of plot<varats.plots.plots.PlotRegistry.plots>`
+                    that will be generated.
+        file_format: The file format of the generated plot.
+        kwargs: Additional arguments that will be passed to the plot class.
+    """
+
+    def __init__(self, name: str, output_path: Path, plot_type: str,
+                 file_format: str, **kwargs: tp.Any) -> None:
+        super(PlotArtefact, self).__init__(ArtefactType.plot, name, output_path)
+        self.__plot_type = plot_type
+        self.__plot_type_class = PlotRegistry.get_class_for_plot_type(plot_type)
+        self.__file_format = file_format
+        self.__plot_kwargs = kwargs
+
+    @property
+    def plot_type(self) -> str:
+        """
+        The :attr:`type of plot<varats.plots.plots.PlotRegistry.plots>` that
+        will be generated.
+        """
+        return self.__plot_type
+
+    @property
+    def plot_type_class(self) -> tp.Type[Plot]:
+        """
+        The class associated with :func:`plot_type`.
+        """
+        return self.__plot_type_class
+
+    @property
+    def file_format(self) -> str:
+        """
+        The file format of the generated plot.
+        """
+        return self.__file_format
+
+    @property
+    def plot_kwargs(self) -> tp.Any:
+        """
+        Additional arguments that will be passed to the plot_type_class.
+        """
+        return self.__plot_kwargs
+
+    def get_dict(self) -> tp.Dict[str, str]:
+        artefact_dict = super().get_dict()
+        artefact_dict['plot_type'] = self.__plot_type
+        artefact_dict['file_format'] = self.__file_format
+        artefact_dict = {**self.__plot_kwargs, **artefact_dict}
+        return artefact_dict
+
+    def generate_artefact(self) -> None:
+        """
+        Generate the specified plot.
+        """
+        plot = self.plot_type_class(**self.plot_kwargs)
+        plot.style = "ggplot"
+        plot.save(self.output_path, self.file_format)
+
+
+class ArtefactType(Enum):
+    """
+    Enum for the different artefact types.
+
+    The name is used in the ``artefacts.yaml`` to identify what kind of artefact
+    is described.
+    The values are tuples ``(artefact_class, version)`` consisting of the class
+    responsible for that kind of artefact and a version number to allow
+    evolution of artefacts.
+    """
+    plot = (PlotArtefact, 1)
+
+
+class Artefacts:
+    """
+    A collection of :class:`Artefact`\ s.
+    """
+
+    def __init__(self, artefacts: tp.Iterable[Artefact]) -> None:
+        self.__artefacts = {artefact.name: artefact for artefact in artefacts}
+
+    @property
+    def artefacts(self) -> tp.Iterable[Artefact]:
+        """
```

Comment:
```suggestion
        r"""
```
https://www.flake8rules.com/rules/W605.html
some python stuff because of escapes